src/transformers/models/resnet/modeling_tf_resnet.py

```diff
@@ -0,0 +1,595 @@
+        window_starts = tf.math.floor((tf.range(output_dim, dtype=tf.float32) * input_dim) / output_dim)
+        window_starts = tf.cast(window_starts, tf.int64)
+        window_ends = tf.math.ceil((tf.range(1, output_dim + 1, dtype=tf.float32) * input_dim) / output_dim)
+        window_ends = tf.cast(window_ends, tf.int64)
+
+        # pool_selector is a boolean array of shape (output_dim,) where 1 indicates that output position
+        # has a big receptive field and 0 indicates that that output position has a small receptive field
+        pool_selector = tf.cast(window_ends - window_starts - small_window, tf.bool)
+
+        # Since we concatenated the small and big pools, we need to do a bit of
+        # pointer arithmetic to get the indices of the big pools
+        small_indices = window_starts
+        big_indices = window_starts + small_pool.shape[axis]
+
+        # Finally, we use the pool_selector to generate a list of indices, one per output position
+        gather_indices = tf.where(pool_selector, big_indices, small_indices)
+
+        # Gathering from those indices yields the final, correct pooling
+        return tf.gather(both_pool, gather_indices, axis=axis)
+
+    def call(self, inputs: tf.Tensor):
+        # We can't perform NCHW operations on CPU. And so transpose our inputs if necessary
+        if self.input_ordering == "NCHW":
+            inputs = tf.transpose(inputs, (0, 2, 3, 1))
+        input_shape = inputs.shape[1:3]
+
+        if input_shape[0] % self.output_dims[0] == 0 and input_shape[1] % self.output_dims[1] == 0:
+            # If we're resizing by an integer factor on both dimensions, we can take
+            # a very quick shortcut.
+            h_resize = int(input_shape[0] // self.output_dims[0])
+            w_resize = int(input_shape[1] // self.output_dims[1])
+
+            outputs = tf.nn.avg_pool2d(
+                inputs,
+                ksize=(h_resize, w_resize),
+                strides=(h_resize, w_resize),
+                padding="VALID",
+                # data_format=self.input_ordering,
+            )
+
+            if self.input_ordering == "NCHW":
+                # NHWC -> NCHW
+                outputs = tf.transpose(outputs, (0, 3, 1, 2))
+            return outputs
+
+        else:
+            # If we can't take the shortcut, we do a 1D pool on each axis
+            h_pooled = self.pseudo_1d_pool(inputs, h_pooling=True)
+            return self.pseudo_1d_pool(h_pooled, h_pooling=False)
+
+
+@keras_serializable
+class TFResNetMainLayer(tf.keras.layers.Layer):
+    config_class = ResNetConfig
+
+    def __init__(self, config: ResNetConfig, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.config = config
+        self.embedder = TFResNetEmbeddings(config, name="embedder")
+        self.encoder = TFResNetEncoder(config, name="encoder")
+        self.pooler = TFAdaptiveAvgPool2D((1, 1), input_ordering="NCHW")
```

Comment:
I just made the same comment in the RegNets PR, but a resize to `(1, 1)` should probably be replaced by the built-in [GlobalAveragePooling2D](https://www.tensorflow.org/api_docs/python/tf/keras/layers/GlobalAveragePooling2D). I'm sorry I didn't inspect the actual calls sooner - we probably don't need that custom op in this model at all!